Nuage/Views/WebView.swift

```diff
@@ -83,4 +76,30 @@ class WebViewCoordinator: NSObject, WKNavigationDelegate, WKUIDelegate {
         webView.window?.close()
     }
     
+    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        if keyPath == "URL" {
+            let webView = object! as! WKWebView
+          
+            if (webView != nil) {
```

Comment:
I'm sure there is some better way here so we don't have to force-unwrap the value and cast it from `Any` to `WKWebView`

Reply:
You can do 
```
if let webView = object as? WKWebView {

}
```

Reply:
This compiles:
```swift
if let webView = object as? WKWebView {
  self.scanCookiesForAccessToken(webView: webView)
}
```
but it's crashing the app. It's crashing in the same why as it did before I added the `if (webView != nil)` check around `self.scanCookiesForAccessToken`:
```swift
let webView = object! as! WKWebView
self.scanCookiesForAccessToken(webView: webView)
```

I only get a `Thread 1: EXC_BAD_ACCESS (code=1, address=0x3294db7d48d0)` which leads me to believe that some of the typing/casting is going wrong.

Reply:
Okay, seems like using `Combine` is more reliable and works better